pkgs/development/tools/comby/default.nix

```diff
@@ -41,9 +51,21 @@ let
     };
 
   combyKernel = mkCombyPackage { pname = "comby-kernel"; };
-in mkCombyPackage {
+  combySemantic = mkCombyPackage { pname = "comby-semantic"; extraBuildInputs = [ ocamlPackages.cohttp-lwt-unix ]; };
+in
+mkCombyPackage {
   pname = "comby";
 
+  # tests have to be removed before building otherwise installPhase will fail
+  # cli tests expect a path to the built binary
+  preBuild = ''
+    substituteInPlace test/common/dune \
+      --replace "test_cli_list" "" \
+      --replace "test_cli_helper" "" \
+      --replace "test_cli" ""
+    rm test/common/{test_cli_list,test_cli_helper,test_cli}.ml
```

Comment:
Why can't we sub in the path of the built binary? We do know that it is going to end up at `$out/bin/comby`. It would be nice to have some basic tests that run the result.

Reply:
The checkPhase runs before the installPhase right ? so at the time of the checkPhase there is nothing in $out/bin/comby . After the buildPhase there is probably a bin in some directory, but I don't know dune well enough to know where it is. Do you think a `TODO: replace with the binary dune build path` would be better ?

Reply:
But if it runs before Nix moves it why can't it find the original binary?

I did originally think that it should work because the binary should be in the usual place, but due to the issue I thought maybe it was something else.

I think I'll merge for now and we can always add a fix to run the tests.

Reply:
you're right of course.
the build phase produces a binary somewhere in a `_build` directory. it's just a matter of testing different paths I think.
the installPhase seems to know where that binary is and move it into the $out.

Reply:
I'm just surprised that the location in the `_build` directory is not the location that the tests expect. I would assume that it is the standard OCaml stuff. Ah well. If you do figure it out I'd be happy to approve a follow-up patch 🙂

Reply:
I've tried a couple more things without success
`../../../_build/default/src/main.exe`

I tried that because there is a `_build/default/src/main.exe`
there is also a `_build/install/default/bin` that symlinks to the first path.

hopefully this is helpful for the next person.